src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,173 @@
+            last = newNode;
+        } else {
+            f.prev = newNode;
+        }
+        size++;
+    }
+
+    public void linkLast(T t) {
+        if (last == null) {
+            Node<T> newNode = new Node<>(null, t, null);
+            first = newNode;
+            last = newNode;
+        } else {
+            Node<T> newNode = new Node<>(last, t, null);
+            last.next = newNode;
+            last = newNode;
+        }
+        size++;
+    }
+
     @Override
     public void add(T value) {
+        linkLast(value);
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Wrong index " + index);
+        }
+        if (index == size) {
+            linkLast(value);
+        } else if (index == 0) {
+            linkFirst(value);
+        } else {
+            Node<T> pred = getNode(index - 1);
+            Node<T> succ = pred.next;
+            Node<T> newNode = new Node<>(pred, value, succ);
+            pred.next = newNode;
+            succ.prev = newNode;
+            size++;
+        }
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T t : list) {
+            add(t);
+        }
+    }
+
+    private Node<T> getNode(int index) {
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Wrong index " + index);
+        }
+        if (index < (size >> 1)) {
+            Node<T> x = first;
+            for (int i = 0; i < index; i++) {
+                x = x.next;
+            }
+            return x;
+        } else {
+            Node<T> x = last;
+            for (int i = size - 1; i > index; i--) {
+                x = x.prev;
+            }
+            return x;
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        return getNode(index).item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        Node<T> node = getNode(index);
+        T oldValue = node.item;
+        node.item = value;
+        return oldValue;
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        Node<T> node = getNode(index);
+        final T oldValue = node.item;
+        removeNode(node);
+        return oldValue;
     }
 
     @Override
     public boolean remove(T object) {
+        if (object == null) {
+            Node<T> x = first;
+            while (x != null) {
+                if (x.item == null) {
+                    removeNode(x);
+                    return true;
+                }
+                x = x.next;
+            }
+        } else {
+            Node<T> x = first;
+            int index = 0;
+            while (x != null) {
+                if (object.equals(x.item)) {
+                    remove(index);
+                    return true;
+                }
+                x = x.next;
+                index++;
+            }
+        }
```

Comment:
remove if-else, merge into one block, use equals condition as in Objects.equals (but you should write your own)